Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -165,7 +165,27 @@ public struct Reducer<State, Action, Environment> {
   ///   state.
   public var optional: Reducer<State?, Action, Environment> {
     .init { state, action, environment in
-      guard state != nil else { return .none }
+      guard state != nil else {
+        assertionFailure(
+          """
+          "\(debugCaseOutput(action))" was received by an optional reducer when its state was \
+          "nil". This can happen for a few reasons:
+
+          * The optional reducer was combined with another reducer that set "\(State.self)" to \
+          "nil" when it handled this action. Combine optional reducers before reducers that can \
+          set their state to "nil". This ensures that optional reducers can handle their actions \
+          while their state is still non-"nil".
+
+          * This action was sent to the store while state was "nil". Make sure that actions for \
+          this reducer can only be sent to a view store when state is non-"nil". In SwiftUI \
+          applications, use "IfLetStore".
+
+          * An effect returned this action while state was "nil". Make sure that effects for this \
+          optional reducer are canceled when optional state is set to "nil".
```

Comment:
```suggestion
          "\(debugCaseOutput(action))" was received by an optional reducer when its state was \
          "nil". This can happen for a few reasons:

          * The optional reducer was combined with another reducer that set "\(State.self)" to \
          "nil" when it handled this action. Combine optional reducers before reducers that can \
          set their state to "nil". This ensures that optional reducers can handle their actions \
          while their state is still non-"nil".

          * An active effect emitted this action while state was "nil". Make sure that effects for this \
          optional reducer are canceled when optional state is set to "nil".

          * This action was sent to the store while state was "nil". Make sure that actions for \
          this reducer can only be sent to a view store when state is non-"nil". In SwiftUI \
          applications, use "IfLetStore".
```